### PR TITLE
feat(ci): add staging cloud smoke test

### PR DESCRIPTION
**Pull request by kentwelcome**
## Summary
- Add daily scheduled staging cloud smoke test that runs after the nightly build (3 AM UTC+8)
- Add `workflow_dispatch` trigger with environment selector (production/staging) for manual runs
- Use `resolve-environment` job to cleanly route between production and staging tests

## Required GitHub Settings
- **Secret**: `RECCE_CLOUD_TOKEN_STAGING` — auth token for staging cloud
- **Variable**: `RECCE_CLOUD_API_HOST_STAGING` — staging cloud API URL

## Test plan
- [ ] Verify push/PR triggers only run production smoke test (unchanged behavior)
- [ ] Verify scheduled trigger only runs staging smoke test
- [ ] Verify manual dispatch with "production" runs production test
- [ ] Verify manual dispatch with "staging" runs staging test
- [ ] Configure `RECCE_CLOUD_TOKEN_STAGING` secret and `RECCE_CLOUD_API_HOST_STAGING` variable in repo settings

Resolves: DRC-3244

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/workflows/integration-tests-cloud.yaml [hunk 1]
```diff
@@ -13,17 +13,46 @@ on:
...
+        required: true
+        options:
+          - production
+          - staging
+        default: production
 
 # Explicitly limit permissions for pull_request_target
 permissions:
   contents: read
   pull-requests: read
```
[the remark this answers is not in the document]
Comment (kentwelcome) [reply .github/workflows/integration-tests-cloud.yaml]: Fixed — updated the comment to remove the stale `pull_request_target` reference.

File: .github/workflows/integration-tests-cloud.yaml [hunk 2]
```diff
@@ -13,17 +13,46 @@ on:
...
+      - name: Resolve target environment
+        id: resolve
+        run: |
+          if [[ "${{ github.event_name }}" == "schedule" ]]; then
+            echo "environment=staging"
+          elif [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            echo "environment=${{ inputs.environment }}"
+          else
+            echo "environment=production"
+          fi >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (kentwelcome) [reply .github/workflows/integration-tests-cloud.yaml]: Good call — applied the suggestion. Each branch now writes to `$GITHUB_OUTPUT` explicitly.

File: .github/workflows/integration-tests-cloud.yaml [hunk 3]
```diff
@@ -91,3 +120,72 @@ jobs:
...
+          fi
+
+      - name: Run smoke test - dbt (staging)
+        run: |
+          source .venv/bin/activate
+          ./integration_tests/dbt/smoke_test_cloud.sh
+        env:
+          GITHUB_TOKEN: ${{ secrets.RECCE_CLOUD_TOKEN_STAGING }}
+          RECCE_STATE_PASSWORD: ${{ vars.RECCE_STATE_PASSWORD }}
+          RECCE_CLOUD_API_HOST: ${{ vars.RECCE_CLOUD_API_HOST_STAGING }}
```
[the remark this answers is not in the document]
Comment (kentwelcome) [reply .github/workflows/integration-tests-cloud.yaml:167]: Acknowledged. The two jobs only differ in env vars (secrets/API host), so the duplication is minimal. Merging them into one job with conditional env vars would add complexity to the conditions we just simplified. Keeping them separate for now — we can extract a reusable workflow if we add more environments later.

**Comment by even-wei** [comment]
# Code Review: PR #1310

**Files reviewed:** 1
**Categories:** Config/infra (CI workflow)
**Passes run:** A, B, D, F

## Validation Results

### Pass A: Correctness & Logic — PASS

- **Cron schedule** `0 19 * * 0,1,2,3,4` correctly maps to Mon–Fri 03:00 UTC+8 (Sun–Thu 19:00 UTC). ✓
- **Environment routing** logic is sound: `schedule` → staging, `workflow_dispatch` → user input, `push`/`pull_request` → production. ✓
- **Job dependency chain** is correct:
  - Staging path: `resolve-environment` → `smoke-test-cloud-staging` (authorize skipped, production smoke-test auto-skipped via GitHub's needs-skip propagation)
  - Production path: `resolve-environment` → `authorize` → `smoke-test-cloud` (staging smoke-test skipped via `if` condition)
- **Pre-existing behavior preserved**: push-to-main and PR triggers still route through authorize → smoke-test-cloud exactly as before, with only the added `resolve-environment` step in the chain. ✓

### Pass B: Security — PASS

- Secrets (`RECCE_CLOUD_TOKEN_STAGING`, `SLACK_WEBHOOK_URL_DEV`) are properly referenced via `${{ secrets.* }}` and passed as env vars to shell scripts (not interpolated inline). ✓
- `workflow_dispatch` can only be triggered by users with repo write access — no authorization bypass risk from the staging path skipping the `authorize` job. ✓
- `inputs.environment` is a `choice` type with fixed options — no injection surface. ✓
- `github.event_name` is GitHub-controlled and limited to a fixed set of values. ✓

### Pass D: Error Handling & Edge Cases — PASS

One note below regarding Slack notification resilience.

### Pass F: Diff-Specific Checks — PASS

- `authorize` and `smoke-test-cloud` both gain the `environment == 'production'` gate — consistent and correct. ✓
- `smoke-test-cloud-staging` correctly mirrors the production job's matrix and install steps while using staging-specific secrets/vars. ✓
- Staging job correctly adds `RECCE_CLOUD_API_HOST` env var (not needed for production since it defaults to the production host). ✓

### Verification Results

YAML syntax valid. No linting issues in the workflow diff.

## Verdict: GO

No blockers or issues found. Clean, well-structured addition.

### Notes
1. **`SLACK_WEBHOOK_URL_DEV` not listed in PR's "Required GitHub Settings"** — the PR body lists `RECCE_CLOUD_TOKEN_STAGING` and `RECCE_CLOUD_API_HOST_STAGING` but doesn't mention `SLACK_WEBHOOK_URL_DEV`. If this secret isn't configured, the `notify-staging-failure` job's `curl` will fail. The existing Slack pattern in `address-dependabot.yaml` guards with `if [ -n "$SLACK_WEBHOOK_URL" ]` before posting — consider adding the same guard here for resilience. (`.github/workflows/integration-tests-cloud.yaml:162`)

---
🤖 Reviewed by Claude Code

**Review by even-wei — APPROVED** [review]
Claude Code Review: No critical issues found. Clean CI workflow addition with correct environment routing logic.
